Makefile

```diff
@@ -1,12 +1,21 @@
 #
-# github.com/docker/cli 
+# github.com/docker/cli
 #
 
+VERSION ?= unknown-version
+GITCOMMIT ?= $(shell git rev-parse --short HEAD 2> /dev/null || true)
+BUILDTIME ?= $(shell date --utc --rfc-3339 ns 2> /dev/null | sed -e 's/ /T/')
+
+LDFLAGS := -X github.com/docker/cli/cli.GitCommit=${GITCOMMIT} \
+	-X github.com/docker/cli/cli.BuildTime=${BUILDTIME} \
+	-X github.com/docker/cli/cli.Version=${VERSION} \
```

Comment:
I'd be pretty happy avoiding all this makefile stuff and just using `script/build/x` with source.

The makefile only needs:

```

build: clean
    script/build/binary

cross: clean
    script/build/cross

```